src/fontman/font_manager.cpp

```diff
@@ -63,19 +79,101 @@ static bool             g_fontManagerIsInit = false;
 typedef std::unordered_map<std::string, int> FontsHash;
 //! Database of available fonts
 static FontsHash        g_fontNameToId;
+//! Backup of the database of available fonts
+static FontsHash        g_fontNameToIdBackup;
+static FontsHash        g_fontNameToIdBackupWorld;
 
 static bool             g_double_pixled = false;
 
+static const int c_smbxFontsMapMax = 100;
```

Comment:
It's a good idea to make a static limit, but I think this should depend on whether scripting is enabled (4 like SMBX if not enabled, 8-16 if enabled).

Reply:
Since we currently have no Lua scripting, we can make it universally 4 until v1.3.7.

Reply:
16 better, 4 is bad, because it kills the ability to add new fonts without replacing of existing (custom fonts were already can be used at the LunaScript in a condition the "font-override" entry had been added to assign the font ID).

Reply:
I tried Font 5, works as well... (Message box untested)

Reply:
Should 8 fonts fine for low-end device?

Reply:
Okay, I tweaked this thing, and also I replaced `int` below with the `vbint_t` that uses int16_t type on low-end devices.